src/app/core/services/office/office.service.ts

```diff
@@ -1,9 +1,61 @@
 import { Injectable } from '@angular/core';
+import { Observable, of, throwError } from 'rxjs';
+import { OfficeModel } from '../../../shared/models/office.model';
+import { ApiService } from '../../http/api/api.service';
+import { catchError, map } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class OfficeService {
 
-  constructor() { }
+  constructor(private apiService: ApiService) { }
+
+  public handleOfficeNames(offices: OfficeModel[]): OfficeModel[] {
+    return Object.keys(offices).map(key => Object.assign({}, {...offices[key], name: key}));
+  }
+
+  public handleSingleOfficeNames(office: OfficeModel, data): OfficeModel {
+    return Object.assign({}, { ...office, ...data });
+  }
```

Comment:
Why Object.assign({}, { ...office, ...data }) and not simple { ...office, ...data }?